src/log_prob.jl

```diff
@@ -0,0 +1,438 @@
+# Functions to compute the log probability of star parameters and galaxy
+# parameters given pixel data
+
+using Distributions
+import ..SensitiveFloats: SensitiveFloat, zero_sensitive_float, clear!
+
+include("bivariate_normals.jl")
+include("log_prob_util.jl")
+EPS = 1e-6
+
+###################################
+# log likelihood function makers  #
+###################################
+
+
+"""
+Creates a vectorized version of the star logpdf as a function of
+unconstrained params.
+
+    star_params = [lnr, lnc1, ..., lnc4, ra, dec]
+
+Args:
+  - images: Vector of TiledImage types (data for log_likelihood)
+  - active_pixels: Vector of ActivePixels on which the log_likelihood is based
+  - ea: ElboArgs book keeping argument
+
+Returns:
+  - star_logpdf  : unnormalized logpdf function handle that takes in a flat,
+                   unconstrained array as parameter
+  - star_logprior: star param log prior function handle that takes in same
+                   flat, unconstrained array as parameter
+"""
+function make_star_logpdf(images::Vector{TiledImage},
+                          active_pixels::Vector{ActivePixel},
+                          ea::ElboArgs)
+
+    # define star prior log probability density function
+    prior    = construct_prior()
+    subprior = prior.star
+
+    function star_logprior(state::Vector{Float64})
+        brightness, colors, u = state[1], state[2:5], state[6:end]
+        return color_logprior(brightness, colors, prior, true)
+    end
+
+    # define star log joint probability density function
+    function star_logpdf(state::Vector{Float64})
+        ll_prior = star_logprior(state)
+        brightness, colors, position = state[1], state[2:5], state[6:end]
+        dummy_gal_shape = [.1, .1, .1, .1]
```

Comment:
Or by making it a tuple--those are stack allocated and vectors are heap allocated.